client/state/themes/actions.js

```diff
@@ -452,16 +452,16 @@ export function installTheme( themeId, siteId ) {
 		}
 
 		return wpcom.undocumented().installThemeOnJetpack( siteId, themeId )
-			.then( () => {
+			.then( ( { id } ) => {
 				// We do not `dispatch( receiveTheme( theme, siteId ) )` here because
 				// in our UI, themes from WP.com (and WP.org) are already present in
 				// a separate list, and we do not want to duplicate them.
-
 				dispatch( {
 					type: THEME_INSTALL_SUCCESS,
 					siteId,
-					themeId
+					themeId: id,
```

Comment:
I think we should add a comment here why we aren't using the `themeId` arg since that'd seem like such an obvious alternative.